modules/user/user.service.js

```diff
@@ -0,0 +1,155 @@
+const { AuthenticationError, UserInputError } = require('apollo-server');
+const bcrypt = require('bcryptjs');
+const User = require('./user.model');
+const {
+  validateRegisterInput,
+  validateLoginInput,
+} = require('../../utils/validateUser');
+const generateToken = require('../../utils/createToken');
+
+class UserService {
+  getAllUsers() {
+    return User.find();
+  }
+
+  getUserById(id, auth) {
+    if (auth.errors) throw new Error(auth.errors);
+    return User.findById(id);
+  }
+
+  async updateUser(id, {
+    firstName, lastName, email, password,
+  }, auth) {
+    if (auth.errors) throw new Error(auth.errors);
```

Comment:
It should be thrown inside the context, not inside service

Reply:
fixed